Change Markdown engine and syntax highlighter

amree opened the pull request:
Based on https://git.io/vgYlB, GitHub will only support kramdown in the future. This will also remove pygments which is Python dependent.

I also need to update all the links which is not formatted in Markdown to make sure it'll be displayed as link.

k4ml commented:
Thanks for looking into this. Is it ready for merge ?

k4ml commented:
Since kramdown support GFM, that mean it also support autolink so we don't have to fix all the links. Need to verify this, though.

amree commented:
It's working the last time I tried in local. Regarding the link, apparently it doesn't work, that's why I formatted those links.

k4ml commented:
Looking at this post, the autolink work - http://devkini.github.io/daily-git-commands/#stash-patch

k4ml commented:
Oh ...

> The page build completed successfully, but returned the following warning:
> 
> You are currently using the 'redcarpet' Markdown engine, which will not be supported on GitHub Pages after May 1st. At that time, your site will use 'kramdown' for markdown rendering instead. To suppress this warning, remove the 'markdown' setting in your site's '_config.yml' file and confirm your site renders as expected. For more information, see https://help.github.com/articles/updating-your-markdown-processor-to-kramdown.
> 
> GitHub Pages was recently upgraded to Jekyll 3.0. It may help to confirm you're using the correct dependencies:
> 
>   https://github.com/blog/2100-github-pages-now-faster-and-simpler-with-jekyll-3-0
> 
> For information on troubleshooting Jekyll see:
> 
>   https://help.github.com/articles/using-jekyll-with-pages#troubleshooting
> 
> If you have any questions you can contact us by replying to this email.

Will test this later today.

amree commented:
The link that you mentioned doesn't work with the new config, but it'll probably work with the old config (same as other links).

P/S: I've rebased my branch to the current master's branch and updated the new link. I can squeeze those commits if you prefer it squeezed.

k4ml commented:
Merged. Thanks !
